Automattic/wp-calypso
# A4A: Updates to get Site Details pane fully functional

## Proposed Changes

- This PR removes the custom tags and tag field implementation in A4A, and instead uses the WordPress `FormTokenField` component that is native to both WordPress and Calypso.
- We also decided to go ahead and get site notes wired up and working.

## Testing Instructions

Load the site tags field for an A4A site (inside site details pane) and check:
- Existing tags load with the page.
- New tags get submitted in the network request.
- Removing tags actually removes them.
- A reload maintains the actual state of the site tags.

Also, while you're there, you might as well test the notes field also:
- Submitting a note works, shows up immediately, and persists upon refresh.
- Styles/design of the notes are roughly okay.

<img width="1728" alt="Screenshot 2024-05-02 at 5 45 48 PM" src="https://github.com/Automattic/wp-calypso/assets/5528445/67af5502-ed6f-498b-b447-29b1c824fad7">

## Pre-merge Checklist
